### ISSUE TITLE
Expression is always true for Array destructuring

### ISSUE THREAD

**Issue by FlorinAsavoaie** [palantir/tslint#4874]
### Bug Report

- __TSLint version__: 5.20.0
- __TypeScript version__: 3.6.3
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
// tslint:disable-next-line:no-magic-numbers
const [beg, end] = "me".split(":", 2);

if (end !== undefined) {
    // tslint:disable-next-line:no-console
    console.log("Got some colons");
}
```

with `tslint.json` configuration:

```json
{
    "extends": ["tslint:all"],
    "rules": {}
}
```

#### Actual behavior
When running tslint from CLI (strange it doesn't happen in WebStorm or on the TSLint PlayGround), it says that *Expression is always true.* at the line with the `if` condition.

#### Expected behavior
As you can see, expression is definitely not always true.

[PlayGroundLink]: https://palantir.github.io/tslint-playground/?saved=N4Igxg9gJgpiBcID0SAEBlALgQwE6dUwE8AHASwDsBzVSwgCxlUllQCMIAPQiVKmAgGcc+GFAB0AHQrSUhQQBtKmeFDKDsbBTAC0FGJ0w6l++BQg6AttipkwegK6W2MXIOmQKw1AG0XVABpUGAooAF1UAF5USRBLGFjxQRIlTAAKWPhYoIAmAEoAbmlpMgAzVDSQqFQAQkjoh1CYUsoxPNRgaVRu1DlMRWVVdU1tPQMjExgzC09BCG0untn5mHEFCCoMkABxCCEIeOZ5iC9YwukAX2kQAPATlqoEZDQATQgHZmwKVEtoMqIGExPA8HLhsJgyCdUIxcEwAO5kTD0VBEd64VAQOEUKQUOQAWWwAGsmIJQUwkeCUWjUAArObfdSoABu2CUUCCX2qZSpH1wjVoFEwEFkaD2MNoglJMEEQUwuABYEYYEJlBoSKBJzm2lQpQg6NcuD1ghx0k63x6sXGVXcCF8sX6JhUrIUsTCAUW3VifO0NvgHQuvTQ2Cg1WDaghJ1ZqG90tQnMBZHRwLIVFB4MhMgoVwoIAuQA

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
